src/atlas/grid/Tiles.h

```diff
@@ -0,0 +1,129 @@
+
+// Forward declarations
+namespace eckit {
+class Parametrisation;
+class Hash;
+}  // namespace eckit
+
+//---------------------------------------------------------------------------------------------------------------------
+
+namespace atlas {
+class PointXY;
+class PointLonLat;
+
+namespace util {
+class Config;
+}  // namespace util
+
+#ifndef DOXYGEN_SHOULD_SKIP_THIS
+namespace cubedspheretiles {
+class CubedSphereTiles;
+class FV3CubedSphereTiles;
+class LFRicCubedSphereTiles;
+}  // namespace cubespheretiles
+#endif
+
+//---------------------------------------------------------------------------------------------------------------------
+
+class CubedSphereTiles : DOXYGEN_HIDE(
+    public util::ObjectHandle<atlas::cubedspheretiles::CubedSphereTiles> ) {
+public:
+    using Spec = util::Config;
+
+public:
+    using Handle::Handle;
+    CubedSphereTiles() = default;
+    CubedSphereTiles( const eckit::Parametrisation& );
+
+    /// Type of the cubed-sphere tiles:
+    std::string type() const;
+
+    std::array<std::array<double,6>,2> xy2abOffsets() const;
+
+    std::array<std::array<double,6>,2> ab2xyOffsets() const;
+
+    void tile0Rotate( double xyz[] ) const;
+
+    void tile1Rotate( double xyz[] ) const;
+
+    void tile2Rotate( double xyz[] ) const;
+
+    void tile3Rotate( double xyz[] ) const;
+
+    void tile4Rotate( double xyz[] ) const;
+
+    void tile5Rotate( double xyz[] ) const;
+
+    void tile0RotateInverse( double xyz[] ) const;
+
+    void tile1RotateInverse( double xyz[] ) const;
+
+    void tile2RotateInverse( double xyz[] ) const;
+
+    void tile3RotateInverse( double xyz[] ) const;
+
+    void tile4RotateInverse( double xyz[] ) const;
+
+    void tile5RotateInverse( double xyz[] ) const;
+
+    idx_t tileFromXY( const double xy[] ) const;
+
+    idx_t tileFromLonLat( const double lonlat[] ) const;
+
+    void enforceXYdomain( double xy[] ) const;
+
+private:
+    /// Output to stream
+    void print( std::ostream& ) const;
+
+    friend std::ostream& operator<<( std::ostream& s, const CubedSphereTiles& cst );
+
+};
+
+
+class FV3CubedSphereTiles : public CubedSphereTiles {
+public:
+    using CubedSphereTiles::CubedSphereTiles;
+    FV3CubedSphereTiles() : CubedSphereTiles() {
+      std::cout << "grid/Tiles.h FV3CubedSphereTiles"  << std::endl;
+    }
+
+    FV3CubedSphereTiles( const CubedSphereTiles& );
+
+private:
+    const ::atlas::cubedspheretiles::FV3CubedSphereTiles* cubedspheretiles_;
+};
+
+class LFRicCubedSphereTiles : public CubedSphereTiles {
+public:
+    using CubedSphereTiles::CubedSphereTiles;
+    LFRicCubedSphereTiles() : CubedSphereTiles() {
+      std::cout << "grid/Tiles.h LFRicCubedSphereTiles"  << std::endl;
+    }
+
+    LFRicCubedSphereTiles( const CubedSphereTiles& );
+
+private:
+    const ::atlas::cubedspheretiles::LFRicCubedSphereTiles* cubedspheretiles_;
+};
+
+
```

Comment:
The classes `atlas::LFricCubedSphereTiles` and `atlas::FV3CubedSphereTiles` bring nothing extra in terms of API from `atlas::CubedSphereTiles` besides perhaps the empty constructor.
I would like to have the implementations to be hidden more, so that it would be enough to just use
```
auto tiles = atlas::CubedSphereTiles("LFric");
```
and/or via an `eckit::Parametrisation`
```
auto tiles = atlas::CubedSphereTiles( atlas::options::type("LFRic") );    // atlas::util::Config argument
```
The first constructor using a type string could just call the second one using Parametrisation.

Reply:
Done - Handles now removed - I can now do 
```
auto tiles = atlas::CubedSphereTiles("LFRicCubedSphereTiles");
```
but not the latter.

Reply:
I noticed that you use "tile type" as configuration. It would be better to just use "type", which is more in line with Atlas, so that a configuration would be
```
grid:
    type: cubedsphere
    N : 32
    tiles: { type: LFRicCubedSphereTiles }
    projection: { type: cubedsphere_equiangular }
    ... ? ...
```
Which brings me also that perhaps for "nice" reading the name of `LFricCubedSphereTiles::static_type()` could just be "cubedsphere_LFRic" (or lowercase) for consistency also with the projections:
```
grid:
    type: cubedsphere
    N : 32
    tiles:      { type: cubedsphere_lfric }
    projection: { type: cubedsphere_equiangular }
    ... ? ...
```

Reply:
@wdeconinck - I can easily do the cosmetic change changing "LFRicCubedSphereTiles" to cubedsphere_lfric. However, I have a problem in that I need the methods that tiles class gives in the CubedSphereProjectionBase class, so that they can be used the equiangular and equidistant projections.  If it was not for the projection class I could add a Tiles class argument to the constructor of CubedSphere class.  Is there a way of getting a subconfiguration from a parameter class?

i.e.
```
grid:
    type: cubedsphere
    N : 32
    projection:
        type: cubedsphere_equiangular
       ...
        tiles:
           type: cubedsphere_lfric
    ... ? ...
```